haystack_experimental/components/image_converters/document_to_image.py

```diff
@@ -0,0 +1,204 @@
+
+        This method processes the input documents, extracting images from supported file formats and converting them
+        to ImageContents. Documents are categorized based on their file type and processing success.
+
+        :param documents: List of Documents to process. Each document should have metadata containing at minimum
+            a 'file_path' key. PDF documents additionally require a 'page_number' key to specify which page to convert.
+
+        :returns:
+            Dictionary containing three lists:
+            - "image_documents": Document objects that were successfully processed and have corresponding ImageContents.
+                Includes both image files and PDF pages.
+            - "image_contents": ImageContents created from the processed documents. These contain base64-encoded image
+                data and metadata. The order corresponds to the `image_documents` list.
+            - "non_image_documents": Document objects that could not be processed as images. This includes unsupported
+                file types, missing files, and documents with insufficient metadata.
+        """
+        if not documents:
+            return {"image_documents": [], "image_contents": [], "non_image_documents": []}
+
+        root_path = self.root_path or ""
+
+        pdf_docs = []
+        image_docs = []
+        non_image_docs = []
+        missing_info_docs = []
+
+        for doc in documents:
+            file_path = doc.meta.get("file_path")
+            if file_path is None or not Path(root_path, file_path).is_file():
+                missing_info_docs.append(doc)
+                continue
+
+            # Store the non-image documents separately
+            mime_type = doc.meta.get("mime_type") or mimetypes.guess_type(file_path)[0]
+            if mime_type not in IMAGE_MIME_TYPES:
+                non_image_docs.append(doc)
+                continue
+
+            # If mimetype is PDF we also need the page number to be able to convert the right page
+            if mime_type == "application/pdf":
+                page_number = doc.meta.get("page_number")
+                if page_number is None:
+                    missing_info_docs.append(doc)
+                else:
+                    pdf_docs.append(doc)
+                continue
+
+            image_docs.append(doc)
+
+        if missing_info_docs:
+            logger.warning(
+                "In DocumentToImageContent {len_missing_file_paths} Documents are either missing a `file_path` "
+                "key in their metadata or the `file_path` does not point to a valid file. "
+                "They will be returned in the `non_image_documents` output.",
+                len_missing_file_paths=len(missing_info_docs),
+            )
+
+        # We de-duplicate the pdf documents because it's possible that the same PDF page is represented by multiple
+        # documents
+        pdf_docs = self._deduplicate(pdf_docs)
```

Comment:
1. I imagine that deduplication is performed to avoid creating multiple duplicate `ImageContent` objects. Right?
2. What happens if the text of a PDF page is split into N chunks? Is it possible that we lose some textual information in this case? I would like to avoid it. **EDIT**: I found a test that covers this exact use case. WDYT? As a user, I would find it unexpected, but maybe there are good reasons...

Reply:
OK with avoiding the deduplication for now.
We can re-evaluate this in the future if there is interest for this feature. (Unless you think it's a must-have).